homeassistant/components/google_cloud/tts.py

```diff
@@ -0,0 +1,212 @@
+    vol.Optional(CONF_ENCODING, default=DEFAULT_ENCODING): cv.string,
+    vol.Optional(CONF_SPEED, default=DEFAULT_SPEED):
+        vol.Range(min=MIN_SPEED, max=MAX_SPEED),
+    vol.Optional(CONF_PITCH, default=DEFAULT_PITCH):
+        vol.Range(min=MIN_PITCH, max=MAX_PITCH),
+    vol.Optional(CONF_GAIN, default=DEFAULT_GAIN):
+        vol.Range(min=MIN_GAIN, max=MAX_GAIN),
+    vol.Optional(CONF_KEY_FILE, default=''): cv.string,
+})
+
+
+async def async_get_engine(hass, config):
+    """Set up Google Cloud TTS component."""
+    key_file = config[CONF_KEY_FILE]
+    if key_file:
+        key_file = hass.config.path(key_file)
+        if not os.path.isfile(key_file):
+            _LOGGER.error(
+                "GOOGLE_APPLICATION_CREDENTIALS file doesn't exist!"
+            )
+            return None
+
+    return GoogleCloudTTSProvider(
+        hass,
+        key_file,
+        config[CONF_LANG],
+        config[CONF_GENDER],
+        config[CONF_VOICE],
+        config[CONF_ENCODING],
+        config[CONF_SPEED],
+        config[CONF_PITCH],
+        config[CONF_GAIN]
+    )
+
+
+class GoogleCloudTTSProvider(Provider):
+    """The Google Cloud TTS API provider."""
+
+    def __init__(
+        self,
+        hass,
+        key_file,
+        lang,
+        gender,
+        voice,
+        encoding,
+        speed,
+        pitch,
+        gain
+    ):
+        """Init Google Cloud TTS service."""
+        self.hass = hass
+        self.name = 'Google Cloud TTS'
+        self._lang = lang
+        self._gender = gender
+        self._voice = voice
+        self._encoding = encoding
+        self._speed = speed
+        self._pitch = pitch
+        self._gain = gain
+
+        if key_file:
+            self._client = texttospeech \
+                .TextToSpeechClient.from_service_account_json(key_file)
+        else:
+            self._client = texttospeech.TextToSpeechClient()
+
+    @property
+    def default_language(self):
+        """Return the default language."""
+        return self._lang
+
+    @property
+    def supported_languages(self):
+        """Return list of supported languages."""
+        return SUPPORTED_LANGUAGES
+
+    @property
+    def supported_options(self):
+        """Return a list of supported options."""
+        return SUPPORTED_OPTIONS
+
+    async def async_get_tts_audio(self, message, language, options=None):
+        """Load TTS from google."""
+        try:
+            with async_timeout.timeout(10, loop=self.hass.loop):
+                _language = language or self._lang
+                _gender = self._gender
+                _voice = self._voice
+                _encoding = self._encoding
+                _speed = self._speed
+                _pitch = self._pitch
+                _gain = self._gain
+
+                if options:
+                    if CONF_GENDER in options:
+                        _gender = options[CONF_GENDER].lower().capitalize()
```

Comment:
you might want to check all of those options are valid

Reply:
Fixed.